pkg/controller/bootstrap/bootstrapsigner.go

```diff
@@ -0,0 +1,291 @@
+				return e.client.Core().Secrets(api.NamespaceSystem).Watch(lo)
+			},
+		},
+		&api.Secret{},
+		options.SecretResync,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc:    func(_ interface{}) { e.signConfigMap() },
+			UpdateFunc: func(_, _ interface{}) { e.signConfigMap() },
+			DeleteFunc: func(_ interface{}) { e.signConfigMap() },
+		},
+	)
+	return e
+}
+
+// Run runs controller loops and returns immediately
+func (e *BootstrapSigner) Run() {
+	if e.stopChan == nil {
+		e.stopChan = make(chan struct{})
+		go e.configMapsController.Run(e.stopChan)
+		go e.secretsController.Run(e.stopChan)
+	}
+}
+
+// Stop gracefully shuts down this controller
+func (e *BootstrapSigner) Stop() {
+	if e.stopChan != nil {
+		close(e.stopChan)
+		e.stopChan = nil
+	}
+}
+
+// signConfigMap computes the signatures on our latest cached objects and writes
+// back if necessary.
+func (e *BootstrapSigner) signConfigMap() {
+	configMaps := e.listConfigMaps()
+
+	for _, origCM := range configMaps {
+		var needUpdate = false
+
+		newCM, err := copyConfigMap(origCM)
+		if err != nil {
+			utilruntime.HandleError(err)
+			continue
+		}
+
+		// First capture the config we are signing
+		content, ok := newCM.Data[kubeConfigKey]
+		if !ok {
+			glog.V(3).Infof("No %s key in %s/%s ConfigMap", kubeConfigKey, origCM.Namespace, origCM.Name)
+			continue
+		}
+
+		// Next remove and save all existing signatures
+		sigs := map[string]string{}
+		for key, value := range newCM.Data {
+			if strings.HasPrefix(key, signaturePrefix) {
+				tokenID := strings.TrimPrefix(key, signaturePrefix)
+				sigs[tokenID] = value
+				delete(newCM.Data, key)
+			}
+		}
+
+		// Now recompute signatures and store them on the new map
+		tokens := e.getTokens()
+		for tokenID, tokenValue := range tokens {
+			sig, err := computeDetachedSig(content, tokenID, tokenValue)
+			if err != nil {
+				utilruntime.HandleError(err)
+			}
+
+			// Check to see if this signature is changed or new.
+			oldSig, _ := sigs[tokenID]
+			if sig != oldSig {
+				needUpdate = true
+			}
+			delete(sigs, tokenID)
+
+			newCM.Data[signaturePrefix+tokenID] = sig
+		}
+
+		// If we have signatures left over we know that some signatures were
+		// removed.  We now need to update the ConfigMap
+		if len(sigs) != 0 {
+			needUpdate = true
+		}
+
+		if needUpdate {
+			e.updateConfigMap(newCM)
+		}
+	}
+}
+
+func (e *BootstrapSigner) updateConfigMap(cm *api.ConfigMap) {
+	for i := 0; i < e.maxRetries; i++ {
+		_, err := e.client.Core().ConfigMaps(cm.Namespace).Update(cm)
+		if apierrors.IsConflict(err) || apierrors.IsNotFound(err) {
+			// if we got a Conflict error, the ConfigMap was updated by someone else,
+			// and we'll get an update notification later. if we got a NotFound error,
+			// the ConfigMap no longer exists, and we don't need to update it.
+			return
+		}
+	}
+}
+
+// listConfigMaps lists all of the cached config maps
+func (e *BootstrapSigner) listConfigMaps() []*api.ConfigMap {
```

Comment:
Unfortunately, Listers take an indexer.  I don't have one of those as I don't need to watch across namespaces.  It seems silly to create an indexer just to avoid writing a 7 line function.